app/code/Magento/Cron/Model/ResourceModel/Schedule.php

```diff
@@ -81,8 +81,7 @@ public function trySetJobUniqueStatusAtomic($scheduleId, $newStatus, $currentSta
                 ['status' => new \Zend_Db_Expr($connection->quote($newStatus))]
             )
             ->where('current.schedule_id = ?', $scheduleId)
-            ->where('current.status = ?', $currentStatus)
-            ->where('existing.schedule_id IS NULL');
+            ->where('current.status = ?', $currentStatus);
```

Comment:
I believe that is not the proper place to fix this issue. Better to implement clearing of an old task in a clean method. I currently work on this improvement so it will be available in one of next release

Reply:
Yes @kandy. I didn't analyze it 100%. That's why i closed the PR.
I like your idea so good luck with that. The most important is to clean schedules with status = "running" and will never be done. That fixed all other problems. The questions is how to decide that ticket is running and will never be finished.

Reply:
In any case, big thank you for contributing to Magento project.